lib/session/messageSession.ts

```diff
@@ -0,0 +1,887 @@
+      throw new Error("'onError' is a required parameter and must be of type 'function'.");
+    }
+    if (!options) options = {};
+    this._isReceivingMessages = true;
+    this.maxConcurrentCallsPerSession = options.maxConcurrentCallsPerSession || 1;
+    this.autoComplete = !!options.autoComplete;
+    this._onMessage = onSessionMessage;
+    this._onError = onError;
+    const connectionId = this._context.namespace.connectionId;
+    if (this._receiver && this._receiver.isOpen()) {
+      const onSessionMessage = async (context: EventContext) => {
+        this._resetTimerOnNewMessageReceived();
+        const bMessage: ServiceBusMessage = new ServiceBusMessage(this._context, context.message!,
+          context.delivery!);
+        try {
+          await this._onMessage(this, bMessage);
+        } catch (err) {
+          const error = translate(err);
+          // Nothing much to do if user's message handler throws. Let us try abandoning the message.
+          if (this.receiveMode === ReceiveMode.peekLock) {
+            try {
+              log.error("[%s] Abandoning the message with id '%s' on the receiver '%s' since " +
+                "an error occured: %O.", connectionId, bMessage.messageId, this.name, error);
+              await bMessage.abandon();
+            } catch (abandonError) {
+              const translatedError = translate(abandonError);
+              log.error("[%s] An error occurred while abandoning the message with id '%s' on the " +
+                "receiver '%s': %O.", connectionId, bMessage.messageId, this.name, translatedError);
+              this._notifyError(translatedError);
+            }
+          }
+          return;
+        }
+
+        // If we've made it this far, then user's message handler completed fine. Let us try
+        // completing the message.
+        if (this.autoComplete && this.receiveMode === ReceiveMode.peekLock) {
+          try {
+            log.messageSession("[%s] Auto completing the message with id '%s' on " +
+              "the receiver '%s'.", connectionId, bMessage.messageId, this.name);
+            await bMessage.complete();
+          } catch (completeError) {
+            const translatedError = translate(completeError);
+            log.error("[%s] An error occurred while completing the message with id '%s' on the " +
+              "receiver '%s': %O.", connectionId, bMessage.messageId, this.name, translatedError);
+            this._notifyError(translatedError);
+          }
+        }
+      };
+      // setting the "message" event listener.
+      this._receiver.on(ReceiverEvents.message, onSessionMessage);
+      // adding credit
+      this._receiver!.setCreditWindow(this.maxConcurrentCallsPerSession);
+      this._receiver!.addCredit(this.maxConcurrentCallsPerSession);
+    } else {
+      this._isReceivingMessages = false;
+      const msg = `MessageSession with sessionId '${this.sessionId}' and name '${this.name}' ` +
```

Comment:
The `recieve` function in both the streamingReceiver.ts and batchReceiver.ts, calls the `_init` function to intialize the receiver when it is not present or it is not open. But here, we throw an error.

Reply:
The behavior for a SessionReceiver is different than a normal MessageReceiver. 
- If the session receiver gets disconnected, then the sdk should not attempt reconnecting the link. It should rather fail and let the user know about it.
- Lock renewal for SessionReceiver happens at the Session level and not for every message in the Session.

Hence the difference in implementation from the MessageReceiver.